rllib/algorithms/tests/test_worker_failures.py

```diff
@@ -880,24 +814,13 @@ def test_long_failure_period_restore_env(self):
             self.assertTrue(result["num_healthy_workers"] == 1)
             # All workers are still not restored, since env are restored.
             self.assertTrue(
-                not any(
-                    ray.get(
-                        [is_recreated(worker) for worker in a.workers.remote_workers()]
-                    )
-                )
+                not any(a.workers.foreach_worker(is_recreated, local_worker=False))
             )
 
             self.assertTrue(result["evaluation"]["num_healthy_workers"] == 1)
             # All eval workers are still not restored, since env are recreated.
             self.assertTrue(
-                not any(
-                    ray.get(
-                        [
-                            is_recreated(worker)
-                            for worker in a.evaluation_workers.remote_workers()
-                        ]
-                    )
-                )
+                not any(a.workers.foreach_worker(is_recreated, local_worker=False))
```

Comment:
Actually, OOC where do we use this api with `local_worker=True`?

Reply:
WorkerSet.foreach_worker()'s default behavior has been going through both local_worker and remote workers. and you couldn't even change it.
I am making things completely compatible for now until next PR.

Reply:
As long as we have a local worker in WorkerSet, we should support NOT looping through it in `foreach_worker`. Let's leave it. You may want to update all weights for example, but NOT the ones from the local worker b/c that one is already updated.